src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Microsoft.CodeAnalysis.Analyzer.Testing.csproj

```diff
@@ -13,12 +13,42 @@
     <PackageTags>Roslyn Analyzer Test Framework Common</PackageTags>
   </PropertyGroup>
 
+  <Choose>
+    <When Condition="'$(TargetFramework)' == 'net452'">
+      <PropertyGroup>
+        <NuGetApiVersion>4.5.3</NuGetApiVersion>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(TargetFramework)' == 'net46'">
+      <PropertyGroup>
+        <NuGetApiVersion>4.9.4</NuGetApiVersion>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$(TargetFramework)' == 'netstandard1.5'">
+      <PropertyGroup>
+        <NuGetApiVersion>4.6.4</NuGetApiVersion>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <NuGetApiVersion>5.3.1</NuGetApiVersion>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
   <ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="2.6.1" />
     <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="1.0.1" />
     <PackageReference Include="Microsoft.VisualBasic" Version="10.0.1" />
     <PackageReference Include="Microsoft.VisualStudio.Composition" Version="15.6.36" />
     <PackageReference Include="System.ValueTuple" Version="4.5.0" />
+
+    <!-- Use PrivateAssets=compile to avoid exposing our NuGet dependencies downstream as public API. -->
+    <PackageReference Include="NuGet.Common" Version="$(NuGetApiVersion)" PrivateAssets="compile" />
+    <PackageReference Include="NuGet.Protocol" Version="$(NuGetApiVersion)" PrivateAssets="compile" />
+    <PackageReference Include="NuGet.Resolver" Version="$(NuGetApiVersion)" PrivateAssets="compile" />
+    <PackageReference Include="NuGet.Packaging" Version="$(NuGetApiVersion)" />
```

Comment:
📝 This package is required at compile time for test code to use `PackageIdentity`.